src/range.rs

```diff
@@ -7,354 +7,458 @@
-    pub fn higher_than(v: impl Into<V>) -> Self {
+    /// Set of all versions higher to some version
+    pub fn strictly_higher_than(v: impl Into<V>) -> Self {
         Self {
-            segments: SmallVec::one((v.into(), None)),
+            segments: SmallVec::one((Excluded(v.into()), Unbounded)),
         }
     }
 
-    /// Set of all versions strictly lower than some version.
+    /// Set of all versions lower to some version
     pub fn strictly_lower_than(v: impl Into<V>) -> Self {
-        let v = v.into();
-        if v == V::lowest() {
-            Self::none()
-        } else {
-            Self {
-                segments: SmallVec::one((V::lowest(), Some(v))),
-            }
+        Self {
+            segments: SmallVec::one((Unbounded, Excluded(v.into()))),
         }
     }
 
-    /// Set of all versions comprised between two given versions.
-    /// The lower bound is included and the higher bound excluded.
-    /// `v1 <= v < v2`.
-    pub fn between(v1: impl Into<V>, v2: impl Into<V>) -> Self {
-        let v1 = v1.into();
-        let v2 = v2.into();
-        if v1 < v2 {
-            Self {
-                segments: SmallVec::one((v1, Some(v2))),
-            }
-        } else {
-            Self::none()
+    /// Set of all versions lower or equal to some version
+    pub fn lower_than(v: impl Into<V>) -> Self {
+        Self {
+            segments: SmallVec::one((Unbounded, Included(v.into()))),
         }
     }
 
-    /// Construct a simple range from anything that impls [RangeBounds] like `v1..v2`.
-    pub fn from_range_bounds<R, IV>(bounds: R) -> Self
-    where
-        R: RangeBounds<IV>,
-        for<'a> &'a IV: Into<V>,
-    {
-        let start = match bounds.start_bound() {
-            Bound::Included(s) => s.into(),
-            Bound::Excluded(s) => s.into().bump(),
-            Bound::Unbounded => V::lowest(),
-        };
-        let end = match bounds.end_bound() {
-            Bound::Included(e) => Some(e.into().bump()),
-            Bound::Excluded(e) => Some(e.into()),
-            Bound::Unbounded => None,
-        };
-        if end.is_some() && end.as_ref() <= Some(&start) {
-            Self::none()
-        } else {
-            Self {
-                segments: SmallVec::one((start, end)),
-            }
+    /// Set of versions greater or equal to `v1` but less than `v2`.
+    pub fn between(v1: impl Into<V>, v2: impl Into<V>) -> Self {
+        Self {
+            segments: SmallVec::one((Included(v1.into()), Excluded(v2.into()))),
         }
     }
 }
 
-// Set operations.
-impl<V: Version> Range<V> {
-    // Negate ##################################################################
+impl<V: Clone> Range<V> {
+    /// Set containing exactly one version
+    pub fn singleton(v: impl Into<V>) -> Self {
+        let v = v.into();
+        Self {
+            segments: SmallVec::one((Included(v.clone()), Included(v))),
+        }
+    }
+
+    /// Set containing all versions expect one
```

Comment:
`except` not `expect`. More generally, does this new method pull its weight vs `singleton(v).complement()`?

Reply:
No, I think you're right. Ill remove it.

Reply:
Done